### PR TITLE
Clean old orphaned docker containers in e2e test

### LINKED ISSUE #27: Clean up docker container and image generated by e2e tests in setup and failure cases
Changes needed in e2e tests now:

If the test execution fails, the docker image and container are not pruned. This leaves orphaned containers running in the system and if you rerun the tests without cleaning them, your most recent test execution will connect to an old container, giving wrong results.
```bash
+ docker run --name cc-aarch64-ubuntu-fastrtps-dashing -td prajaktg/aarch64-ubuntu-fastrtps-dashing:latest /bin/bash
docker: Error response from daemon: Conflict. The container name "/cc-aarch64-ubuntu-fastrtps-dashing" is already in use by container "d1178d7490b163501567b8205ce09c526c445327179e6b12b5f33611651d18be". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
``` 

1. In the test setup, make sure container and image of the same name as produced by the test don't exist. If they do, remove them.
2. If there are errors in test execution, make sure generated images and containers get cleaned up before the test exits.

_Originally posted by @prajakta-gokhale in https://github.com/aws-ros-dev/cross_compile/pull/26#issuecomment-540241723_

### PR DESCRIPTION
**Pull request by prajakta-gokhale**
Tested to make sure there were running containers before this change and no running containers after this change.

Closes https://github.com/aws-ros-dev/cross_compile/issues/27.

Signed-off-by: Prajakta Gokhale <prajaktg@amazon.com>

### CODE REVIEW

File: test/run_e2e_test.sh [hunk 1]
```diff
@@ -58,6 +60,10 @@ setup(){
   # Copy QEMU binaries
   mkdir -p "$temp_path/sysroot/qemu-user-static"
   cp /usr/bin/qemu-* "$temp_path/sysroot/qemu-user-static"
+
+  if [[ $(docker inspect -f "{{.State.Running}}" "$CONTAINER_NAME") == "true" ]]; then
+    docker stop -t 2 "$CONTAINER_NAME"
```
Comment (thomas-moulard) [inline test/run_e2e_test.sh:71]: we should log a warning here too.

**Review by thomas-moulard — APPROVED** [review]
LGTM with one comment. Thanks for the patch! As discussed, it'd be better in the future to semi-randomize the container name so that we can detect those issues more easily.

**Comment by thomas-moulard** [comment]
Also, you may want to try: https://help.github.com/en/articles/closing-issues-using-keywords instead of commenting on the issue every time.

**Review by piraka9011 — APPROVED** [review]
LGTM

**Pull request merged**